src/ServiceInsight/SequenceDiagram/ModelCreator.cs

```diff
@@ -98,6 +99,19 @@ void Initialize()
 
                 sendingHandler.Out = sendingHandler.Out.Concat(new[] { arrow }).OrderBy(a => a).ToList();
             }
+
+            ReOrderHandlers(firstOrderHandlers);
+        }
+
+        private void ReOrderHandlers(IList<Handler> orderedHandlers)
```

Comment:
@adamralph @distantcam as discussed, for now I have merged the ordering method with initialize.